src/repr/src/row.rs

```diff
@@ -713,6 +726,34 @@ where
 // public api
 
 impl Row {
+    /// Allocate an empty `Row` with a pre-allocated capacity.
+    #[inline]
+    pub fn with_capacity(cap: usize) -> Self {
+        Self {
+            data: SmallVec::with_capacity(cap),
+        }
+    }
+
+    /// Creates a new row from supplied bytes.
+    ///
+    /// # Safety
+    ///
+    /// This method relies on `data` being an appropriate row encoding, and can
+    /// result in unsafety if this is not the case.
+    pub unsafe fn from_bytes_unchecked(data: Vec<u8>) -> Self {
+        Row { data: data.into() }
+    }
+
+    /// Constructs a [`RowPacker`] that will pack datums into this row's
+    /// allocation.
+    ///
+    /// This method clears the existing contents of the row, but retains the
+    /// allocation.
+    pub fn packer(&mut self) -> RowPacker {
```

Comment:
I think making the lifetimes explicit here is good
```suggestion
    pub fn packer<'row>(&'row mut self) -> RowPacker<'row> {
```

Reply:
Updated the signature to use the anonymous lifetime in `RowPacker`.